addons/crm/models/crm_lead.py

```diff
@@ -292,7 +292,7 @@ def _compute_company_id(self):
             # propose a new company based on responsible, limited by team
             if not proposal:
                 if lead.user_id:
-                    proposal = lead.team_id.company_id or lead.user_id.company_id
+                    proposal = lead.team_id.company_id or (lead.user_id.company_id if lead.user_id.company_id in self.env.companies else False)
```

Comment:
Let's try something shorter and more readable 
```suggestion
                    user_company = lead.user_id.company_id.intersection(self.env.companies)
                    proposal = lead.team_id.company_id or user_company
```

Reply:
Done :+1: